### PR TITLE
fix: Fetch all new messages after delta

### PR DESCRIPTION
**Pull request by Ambrdctr**

### CODE REVIEW

File: MailCore/Utils/Array+Extension.swift [hunk 1]
```diff
@@ -50,3 +50,9 @@ public extension LazyMapSequence {
         return Array(self)
     }
 }
+
+public extension Set {
```
Comment (adrien-coye) [inline MailCore/Utils/Array+Extension.swift:54]: move to core ?

Comment (adrien-coye) [reply MailCore/Utils/Array+Extension.swift:54]: More specificaly, can we move the base type extensionn  to core ?
The specific Realm code ca stay here.

public extension List {
public extension LazyFilterSequence {
public extension LazyMapSequence {

File: MailCore/Cache/MailboxManager.swift [hunk 2]
```diff
@@ -699,16 +703,24 @@ public class MailboxManager: ObservableObject {
...
-            .shortUid?.toString() else { return false }
+        var paginationInfo: PaginationInfo?
+
+        if let offset = realm.objects(Message.self).where({ $0.folderId == folder.id })
+            .sorted(by: {
+                if direction == .following {
+                    return $0.shortUid! > $1.shortUid!
+                }
+                return $0.shortUid! < $1.shortUid!
+            }).first?.shortUid?.toString(), let direction {
```
Comment (valentinperignon) [inline MailCore/Cache/MailboxManager.swift:716]:
```suggestion
            }).first?.shortUid?.toString(),
     let direction {
```

File: MailCore/Utils/Array+Extension.swift (hunk 1 above)
Comment (adrien-coye) [reply MailCore/Utils/Array+Extension.swift:54]: discussed, will be done in another PR

**Review by valentinperignon — APPROVED** [review]

**Pull request merged**